src/librustc_errors/emitter.rs

```diff
@@ -785,12 +792,72 @@ fn stderr_isatty() -> bool {
 
 pub enum Destination {
     Terminal(Box<term::StderrTerminal>),
+    BufferedTerminal(Box<BufferedStderr>),
     Raw(Box<Write + Send>),
 }
 
+pub struct BufferedWriter {
+    buffer: Vec<u8>,
+}
+
+impl BufferedWriter {
+    fn _new() -> BufferedWriter {
```

Comment:
Could you add a comment why this has a `_` in front, as well as a comment on the struct itself indicating what it's being used for?